library/src/main/java/com/liulishuo/filedownloader/services/BaseFileServiceUIGuard.java

```diff
@@ -45,6 +45,7 @@
     private final CALLBACK callback;
     private volatile INTERFACE service;
     private final Class<?> serviceClass;
+    protected boolean madeServiceForeground = false;
```

Comment:
1. Prevent redundant operation of `stop`: set to `false` with connecting release.
2. Naming: please change to `runServiceForeground`, and public method to `isRunServiceForeground(void):boolean`

The same to the `FileDownloadServiceSharedTransmit.java` and proxy.

Reply:
set to `false` during `stop` is necessary, because `FileDownloadList#remove` will be invoked many times, so the `stopForeground` may be invoked many times.